company/strategy.md

```diff
@@ -1,36 +1,90 @@
-# Sourcegraph strategy
+# Sourcegraph master plan
+
+Everything we do stems from our [purpose](#purpose), to make it so **everyone can code**. We'll accomplish that over the next 30 years by following our strategy:
+
+- 30 years
+  - [Purpose](#purpose)
+  - [Mission](#mission)
+  - [Values](#values)
+  - [Big Code](#big-code)
+- 3 years
+  - [Principles](#principles)
+  - [Assumptions](#assumptions)
+- 1 year
+  - [1-year vision](#1-year-vision)
+  - [Product direction](#product-direction)
+  - [OKRs](#okrs)
 
 ## Purpose
 
-Sourcegraph exists to create a world where everyone builds software. A world where everyone, not just ~0.2% of the world's population (people who can code today), builds software will see faster and more broadly beneficial technological progress.
+We want to make it so **everyone can code**. A world where everyone, not just ~0.1% of the world population, can code will see faster and more broadly beneficial technological progress.
```

Comment:
I think this document is missing a more coherent bridge between the 1-year vision (make it so existing teams of software professionals can tackle Big Code) and the 30-year purpose (make it so everyone can code). Part of the issue is the rephrasing of the purpose from "everyone builds software" to "everyone can code". The latter has more of an elementary feel ("hey, you wrote 'Hello World' in Java, congrats you can code!") and seems more natural for a company that's tackling the problem of programming education. I believe our goal is not merely to make it so everyone can code, but to foster a world where everyone can *and does* contribute to software used by others. I.e., to make it so everyone can be a producer/contributor to the code economy (alternative phrasing: make it so the code economy is accessible to all). To me, there is a clearer path from "make it so software professions can tackle Big Code" to "make it so everyone can build software / partake in the code economy" than to "make it so everyone can code."

Another thing this document could use is some treatment of the topic of different skill levels / domain expertise in programming itself (and how we help each "level up"). This would help reinforce the point that we are trying to make code accessible to everyone, including groups of different skill levels, spanning from expert engineers to novices to the currently non-technical. But that might be beyond the scope of this PR.

Reply:
Agree on the missing bridge. I plan to do that later (the PR description mentions this). You ok with merging this now and me adding the other parts later (will work with you on them)?

Reply:
@beyang ping

Reply:
Yep, SGTM.